src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java

```diff
@@ -24,43 +34,69 @@
-        "accountuuid" }) )
+@Table(name = "object", uniqueConstraints = @UniqueConstraint(columnNames = { "objectUrn", "accountUuid" }) )
 public class ObjectEntity implements Serializable {
 
+    private static final int UUID_LENGTH = 16;
+    private static final int OBJECT_URN_LENGTH = 767;
+    private static final int TYPE_LENGTH = 255;
+    private static final int MONIKER_LENGTH = 2048;
+    private static final int NAME_LENGTH = 255;
+    private static final int DESCRIPTION_LENGTH = 1024;
+
+    /*
+        Without setting an appropriate Hibernate naming strategy, the column names specified in the @Column annotations below will be converted
+        from camel case to underscore, e.g.: systemUuid -> system_uuid
+
+        To avoid that, select the "old" naming strategy org.hibernate.cfg.EJB3NamingStrategy in your configuration (smartcosmos-ext-objects-rdao.yml):
+        jpa.hibernate.naming_strategy: org.hibernate.cfg.EJB3NamingStrategy
+     */
+
     @Id
     @GeneratedValue(generator = "uuid2")
     @GenericGenerator(name = "uuid2", strategy = "uuid2")
-    @Column(name = "systemuuid", columnDefinition = "BINARY(16)")
+    @Type(type = "uuid-binary")
+    @Column(name = "systemUuid", length = UUID_LENGTH)
     private UUID id;
 
     @NotNull
-    @Column(name="objecturn", nullable = false, updatable = false)
+    @Size(min = 1, max = OBJECT_URN_LENGTH)
+    @Column(name="objectUrn", length = OBJECT_URN_LENGTH, nullable = false, updatable = false)
     private String objectUrn;
 
     @NotNull
+    @Size(min = 1, max = TYPE_LENGTH)
+    @Column(name = "type", length = TYPE_LENGTH, nullable = false)
     private String type;
 
     @NotNull
-    @Column(name = "accountuuid")
+    @Type(type = "uuid-binary")
+    @Column(name = "accountUuid", length = UUID_LENGTH, nullable = false, updatable = false)
     private UUID accountUrn;
 
     @CreatedDate
     @Column(name = "created", insertable = true, updatable = false)
-    @Temporal(TemporalType.TIMESTAMP)
-    private Date created;
+    private Long created;
 
-    @Temporal(TemporalType.TIMESTAMP)
     @LastModifiedDate
-    @Column(name = "lastmodifiedtimestamp", insertable = false, updatable = true)
-    private Date lastModified;
+    @Column(name = "lastModifiedTimestamp", insertable = false, updatable = true)
+    private Long lastModified;
 
+    @Size(max = MONIKER_LENGTH)
+    @Column(name = "moniker", length = MONIKER_LENGTH, nullable = true, updatable = true)
     private String moniker;
 
+    @NotNull
+    @Size(min = 1, max = NAME_LENGTH)
+    @Column(name = "name", length = NAME_LENGTH, nullable = false)
     private String name;
 
+    @Size(max = DESCRIPTION_LENGTH)
+    @Column(name = "description", length = DESCRIPTION_LENGTH)
     private String description;
 
-    @Column(name="activeflag")
+    @Basic
+    @NotNull
+    @Column(name="activeFlag", nullable = false)
     private Boolean activeFlag;
```

Comment:
Lombok just creates getters like `isActive()` for `boolean` fields, but not for `Boolean`. Meh.
https://projectlombok.org/features/GetterSetter.html